src/dataflow/src/source/postgres.rs

```diff
@@ -409,10 +458,67 @@ impl PostgresSourceReader {
                             }
                             self.metrics.lsn.set(self.lsn.into());
                         }
-                        Origin(_) | Relation(_) | Type(_) => {
+                        Relation(relation) => {
+                            let rel_id = relation.rel_id();
+                            if let Some(source_table) = self.source_tables.get(&rel_id) {
+                                // Start with the cheapest check first, this will catch the majority of alters
+                                if source_table.columns.len() != relation.columns().len() {
+                                    error!(
+                                        "alter table detected on {} with id {}",
+                                        source_table.name, source_table.relation_id
+                                    );
+                                    return Err(Fatal(anyhow!(
+                                        "source table {} with oid {} has been altered",
+                                        source_table.name,
+                                        source_table.relation_id
+                                    )));
+                                }
+                                let cols = relation.columns().iter().map(|c| PostgresColumn {
+                                    name: c.name().unwrap().to_string(),
+                                    type_oid: c.type_id(),
+                                    type_mod: c.type_modifier(),
+                                    nullable: bool::default(),
```

Comment:
since this isn't generic and `bool` can be written as a literal these could be clearer with just `false`